ivy/src/main/scala/sbt/IvyInterface.scala

```diff
@@ -21,9 +22,9 @@ final case class ModuleInfo(nameFormal: String, description: String = "", homepa
 }
 
 /** Basic SCM information for a project module */
-final case class ScmInfo(browseUrl: URL, connection: String, devConnection: Option[String] = None)
+final case class ScmInfo(browseUrl: URL, connection: String, devConnection: Option[String] = None, tag: Option[String] = None)
```

Comment:
If you're okay with it and there isn't another solution, breaking it up in regular class + two apply/unapply methods would be the best course IMO (cleaner, no 'clever encoding'...)